databases/relational-databases.md

```diff
@@ -39,69 +77,132 @@ To understand, query, and insert information into a relational database, a techn
 - Users of databases
   - Front end people should have concepts and write very basic SQL
   - Back end people spend a large amount of time with databases
-- Structure of a relational database; tables with columns and references to other tables
+- [Structure of a relational database](https://www.lucidchart.com/pages/database-diagram/database-design)
 - Few basic column types
   - `varchar`
   - `text`
   - `integer`
   - `primary key`, `auto increment`
     - exact type varies depending on database type
-- General SQL statement pattern (verb-subject-optional modifiers)
-- The four basic SQL verbs and their use
+- The [four basic SQL verbs](https://blog.sqlauthority.com/2008/05/13/sql-server-four-basic-sql-statements-sql-operations/) and their use
   - `SELECT`
-    - without modifiers
-    - selecting specific columns: \*, specific columns, AS column aliasing
+    - The SELECT statement is used to select data from a database.
+    - selecting specific columns: \*, specific columns, AS column aliasing. Below are the syntax for the same.
+        ```sql
+        SELECT * FROM table_name;
+        SELECT column1, column2, ...
+        FROM table_name;
+        ```
     - modifier constraints:
       - `WHERE`
+        ```sql
+         SELECT column1, column2, ...    
+         FROM table_name  
+         WHERE condition;
+         ```
       - `ORDER BY`
+        ```sql
+          SELECT column1, column2, ...
+          FROM table_name
+          ORDER BY column1, column2, ... ASC|DESC;
+          ```
       - `LIMIT`
+        ```sql
+         SELECT * FROM table_name
+            LIMIT 3;
+         ```
       - `JOIN`
-        - Should only cover `LEFT INNER/NATURAL/JOINS` with resources pointing to other types
+        - A JOIN clause is used to combine rows from two or more tables, based on a related column between them
+        - Here are the different types of the JOINs in SQL:
+           - (INNER) JOIN: Returns records that have matching values in both tables
+           - LEFT (OUTER) JOIN: Return all records from the left table, and the matched records from the right table
+           - RIGHT (OUTER) JOIN: Return all records from the right table, and the matched records from the left table
+           - FULL (OUTER) JOIN: Return all records when there is a match in either left or right table
   - `INSERT INTO`
     - without modifiers (entire table)
+      ```sql
+       INSERT INTO table_name
+        VALUES (value1, value2, value3, ...);
+          ```
     - with modifier constraints (same as `SELECT`)
+      ```sql
+       INSERT INTO table_name (column1, column2, column3, ...)
+        VALUES (value1, value2, value3, ...);
+        ```
     - from another table `(INSERT INTO a SELECT x, y, z from b)`
     - from another table with constraints `(INSERT INTO a SELECT x, y, z FROM b WHERE x = 'a thing')`
     - Would be good to demonstrate type mismatch (`INSERT` '1' into an integer column), but if using MySQL
```

Comment:
can you rephrase this?  This outline will be read directly by the apprentices

Reply:
@alodahl are you talking about this change? I have removed this line as I didn't feel it important to be here. I am sorry to say but I couldn't find out which change are you talking about. I will be grateful to you if you mention them.

Reply:
- Would be good to demonstrate type mismatch (`INSERT` '1' into an integer column), but if using MySQL to teach this doesn't work as MySQL automatically type casts depending on server config

Reply:
Shall I rephase this? Can I remove this from outline?